### PR TITLE
Laravel 10 support

### PR DESCRIPTION
**Pull request by manavo**
Hi,

First time contributing so sorry if it's not quite right!

Tried to get everything passing in Actions, and hopefully all the edits are correct. Just trying to add Laravel 10 support.

Happy to make any changes if needed. Also changing some dependencies might mean this has to be a new major version? Either way, I've worked off the `3.x` branch.

Thanks!

### CODE REVIEW

**Comment by aglipanci** [comment]
@manavo I am afraid your PR will break support for PHP8.0 which doesn't have to be for Laravel 8 and 9.

**Comment by manavo** [comment]
Laravel 10 has PHP 8.1 as a minimum requirement, so I'm not sure what other options there are?

**Comment by aglipanci** [comment]
@manavo that is true indeed, however the package is support to work for Laravel 8 up to 10, that means that for older versions of Laravel only the older version of the package can be used. Anyway let's see what the maintainers have to say as I haven't actually tested myself.

Great work on the PR anyway!

**Comment by manavo** [comment]
That's why I'm guessing that it might have to be a new major version. Although Laravel 8 and 9 should work fine on PHP 8.1, right? So maybe that's still an OK middle ground.

**Comment by aglipanci** [comment]
Yes they should work fine with 8.1 but that doesn't mean that everyone is using them with the latest version of PHP though. There might be other reasons that people might be using PHP 8.0. If there will be a major version this should work fine.

Anyway I am just assuming :D

**Comment by manavo** [comment]
We'll leave those big decisions for @danharrin 😁

**Comment by danharrin** [comment]
Please retain PHP 8.0 support for now, as we still support Laravel 9

**Comment by manavo** [comment]
Hi @danharrin, not sure that's possible, with Laravel 10 requiring PHP 8.1?

**Comment by danharrin** [comment]
Composer will handle it based on the Laravel version.

**Comment by manavo** [comment]
I updated the `composer.json` file to allow PHP 8.0, if that's what you meant? And I guess it'll resolve whatever is required according to that?

**Comment by aglipanci** [comment]
@manavo revert lines like this as well: https://github.com/squirephp/squire/pull/59/files#diff-b2f45ba596f678a8771422eaa427e1847082d903afd19efc63f75cbfb20243aaR20

**Comment by manavo** [comment]
Good catch, thanks @aglipanci, just updated the packages too!

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -10,10 +10,12 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                php: [8.1, '8.0']
-                laravel: [9.*, 8.*]
+                php: [8.2, 8.1]
```
Comment (aglipanci) [inline .github/workflows/tests.yml]: @manavo bring 8.0 back here as well :)

Comment (manavo) [reply .github/workflows/tests.yml]: Not sure how to do that, without the know failure case of Laravel 10 and PHP 8?

Comment (aglipanci) [reply .github/workflows/tests.yml]: you can add an exclude parameter on the matrix to exclude combinations that fail

```
exclude:
      - php: 8.0
        laravel: 10.*
```

Comment (manavo) [reply .github/workflows/tests.yml]: Got it :)

**Comment by aglipanci** [comment]
Awesome @manavo !

@danharrin you can try to run the workflows and review whenever you can :)

**Comment by danharrin** [comment]
Thanks @manavo and @aglipanci!

**Pull request merged**